src/graphics/engine/text.h

```diff
@@ -159,17 +159,31 @@ enum FontMask
  * \struct UTF8Char
  * \brief UTF-8 character in font cache
  *
- * Only 3-byte chars are supported
+ * 4-bytes chars are supported
  */
 struct UTF8Char
 {
-    char c1, c2, c3;
-    // Padding for 4-byte alignment
-    // It also seems to fix some problems reported by valgrind
-    char pad;
+    char c1, c2, c3, c4;    // note: 4-byte alignment
+    // Padding for 4-byte alignment also seems to fix some problems reported by valgrind
+
+    explicit UTF8Char(const char ch1 = '\0',
+                      const char ch2 = '\0',
+                      const char ch3 = '\0',
+                      const char ch4 = '\0')
+        : c1(ch1), c2(ch2), c3(ch3), c4(ch4)
+    {
+    }
+
+    explicit UTF8Char(const std::string &text,
```

Comment:
`Struct 'UTF8Char' field 'c4' remains uninitialized in constructor`
Same for c3, c2 and c1.